Src/GoogleApis.Core/Apis/Http/ConfigurableMessageHandler.cs

```diff
@@ -64,22 +74,115 @@ public class ConfigurableMessageHandler : DelegatingHandler
         private readonly IList<IHttpExecuteInterceptor> executeInterceptors =
             new List<IHttpExecuteInterceptor>();
 
-        /// <summary>Gets a list of <see cref="IHttpUnsuccessfulResponseHandler"/>.</summary>
-        public IList<IHttpUnsuccessfulResponseHandler> UnsuccessfulResponseHandlers
+        /// <summary>
+        /// Gets a copied list of <see cref="IHttpUnsuccessfulResponseHandler"/>.
+        /// <remarks>
+        /// Since version 1.10, the return value had been changed from <c>IList</c> to <c>IEnumerable</c> in order to
+        /// keep this class thread-safe. The returned list is also a copied version of the handlers. More info is
+        /// available on <a href="https://github.com/google/google-api-dotnet-client/issues/592">Issue 592</a>.
+        /// </remarks>
+        /// </summary>
+        public IEnumerable<IHttpUnsuccessfulResponseHandler> UnsuccessfulResponseHandlers
+        {
+            get
+            {
+                lock (unsuccessfulResponseHandlersLock)

```

Comment:
I think that we should actually keep the 3, each one for each list. I don't want to share one object to lock 3 different lists, it doesn't make sense, and it might actually affect performance.

Regarding comments, Do you think that I should clarify it even further?

Thanks